evaluator.questionnaire/src/main/java/org/opencds/cqf/cql/evaluator/questionnaire/dstu3/QuestionnaireProcessor.java

```diff
@@ -110,9 +142,21 @@ public void processResponseItems(List<QuestionnaireItemComponent> items, List<Qu
     }
 
     @Override
-    public Questionnaire generateQuestionnaire() {
-        var retVal = new Questionnaire();
+    public Questionnaire generateQuestionnaire(String theId, String patientId, IBaseParameters parameters, IBaseBundle bundle, IBaseResource dataEndpoint, IBaseResource contentEndpoint, IBaseResource terminologyEndpoint) {
+        this.patientId = patientId;
+        this.parameters = parameters;
+        this.bundle = bundle;
+        this.dataEndpoint = dataEndpoint;
+        this.contentEndpoint = contentEndpoint;
+        this.terminologyEndpoint = terminologyEndpoint;
```

Comment:
suggestion (non-blocking): I see in some of the other functions you also pass in patientId, bundle, etc. It seems a little confusing to have some functions set state (that is presumably reused at some other point) while others take state provided as an argument. Certainly if we were in a threaded scenario this would break. Would it be simpler to create a `ApplyRequestContext` object and pass that around?

Reply:
Yes, I had a similar thought while working through some issues yesterday but didn't want to climb into that rabbit hole just yet.  Perhaps we create a task for this for next sprint.